crates/adapters/src/transport/s3/mod.rs

```diff
@@ -23,10 +24,28 @@ pub struct S3InputEndpoint {
 }
 
 impl S3InputEndpoint {
-    pub fn new(config: S3InputConfig) -> Self {
-        Self {
-            config: Arc::new(config),
+    pub fn new(config: S3InputConfig) -> AnyResult<Self> {
+        if !config.no_sign_request {
+            if config.aws_access_key_id.is_none() {
+                bail!("the 'aws_access_key_id' property is missing; either set it or use 'no_sign_request=true' to bypass AWS authentication for a public bucket");
+            }
+
+            if config.aws_secret_access_key.is_none() {
+                bail!("the 'aws_secret_access_key' property is missing; either set it or use 'no_sign_request=true' to bypass AWS authentication for a public bucket");
+            }
+        }
+
+        if config.key.is_none() && config.prefix.is_none() {
+            bail!("either 'key' or 'prefix' property must be specified");
```

Comment:
will it be clear from the error that this is about S3?

Reply:
Yes, it shows connector name, which includes table name:
![image](https://github.com/user-attachments/assets/1e14ffa5-9ae4-4c45-9d62-ec95f940651e)

Reply:
Note for later, having the error message say "input connector" instead of endpoint would be nicer (endpoint is an internal term).